File: LES_SERIE_S1_IA_FS_ALGO_PYTHON_C/Série_de_TD-TP_4/exercice8.py
T=[None for i in range(100)]
while True:
    n=int(input("saisir la taille de tableau :  "))
    if(n<100 and n>=1):
        break 
for i in range(0,n):
    T[i]=int(input(f"saisir l'entier numero {i+1}: "))
print("le tableau avant l'inverser: ")
for i in range(0,n):
    print(f" {T[i]} ",end="")
N=n-1
for i in range(0,n-1):
    for j in range(0,N):
        T[j],T[j+1]=T[j+1],T[j]
    N=N-1
print("\nle tableau avant l'inverser: ")
for i in range(0,n):
    print(f" {T[i]} ",end="")
print("")

File: LES_SERIE_S1_IA_FS_ALGO_PYTHON_C/Série_de_TD-TP_4/exercice8.c
#include <stdio.h>
int n,N,temp,T[100];
int main(){
    do{
        printf("saisir la taille de tableau n<100: ");
        scanf("%d",&n);
    }while(n>=100 || n<1);
    //saisir le tableau 
    for(int i=0;i<n;i++){
        printf("saisir l'entier numero %d : ",i+1);
        scanf("%d",&T[i]);
    }
    //affiche le tableau avant d'inverser 
    printf("le tableau avant l'inverser: \n");
    for(int i=0;i<n;i++){
        printf(" %d ",T[i]);
    }  
    printf("\n");
    //inverser et affiche 
    N=n-1;
    for(int i=0;i<n-1;i++){
        for(int j=0;j<N;j++){
            temp=T[j];
            T[j]=T[j+1];
            T[j+1]=temp;
        }
        N--;
    }
    printf("le tableau apres l'inverser: \n");
    for(int i=0;i<n;i++){
        printf(" %d ",T[i]);
    }
    printf("\n");
    return 0;
} 